packages/backend/src/utils/downloader.ts

```diff
@@ -117,7 +117,33 @@ export class Downloader {
   }
 
   private followRedirects(url: string, callback: (message: { ok?: boolean; error?: string }) => void) {
-    const file = fs.createWriteStream(this.target);
+    const tmpFile = `${this.target}.tmp`;
+    const stream = createWriteStream(tmpFile);
+
+    stream.on('finish', () => {
+      stream.close();
+      // Rename from tmp to expected file name.
+      promises
+        .rename(tmpFile, this.target)
+        .then(() => {
+          callback({ ok: true });
+        })
+        .catch((err: unknown) => {
+          callback({ error: `Something went wrong while trying to rename downloaded file: ${String(err)}.` });
+        })
+        .finally(() => {
+          // Finally delete the tmp file
+          promises.rm(tmpFile).catch((err: unknown) => {
+            console.error('Something went wrong while trying to delete the temporary file', err);
+          });
+        });
+    });
+    stream.on('error', e => {
+      callback({
+        error: e.message,
+      });
+    });
```

Comment:
I don't know why but it loose my comment, I was saying that it would have been more readable to me extracting the code to an async function
```
async renameFile(): Promise<void> {
try {

} catch {

} finally {

}
```
and doing in the stream.on('finish')
```ts
   renameFile(..).catch...
```

Reply:
I was almost scared when I did not see any comments and just an approval !!

Reply:
And I just noticed, I am renaming, not copy past, I do not need to delete....